### DISCUSSION TITLE
Is there a way to view all the data contained in a single specific tick?

### DISCUSSION THREAD

**Discussion by pabertay** [saul/demofile#394] [Questions] [answered]
My searches seem to suggest this isn't something many are interested in. Maybe because it's foolish and I lack understanding!

I'm aware I can iterate through individual ticks using `demoFile.on('tickend', e => {});`

Currently, i am using something simple:

```
demoFile.on('tickend', e => {
    if (demoFile.currentTick == targetTick){
        console.log(demoFile.currentTick)
});
```

Just to prove to myself that I'm looking at that one tick. But how can I get all the data that's available for that tick, like a snapshot of the server at that instance?

Something like:

```
demoFile.on('tickend', e => {
    if (demoFile.currentTick == targetTick){
        console.log(JSON.stringify(???))
});
```

Where the ??? is some object representing all the tick data.

**Comment by saul** [comment, marked as answer]
Your best bet is to iterate through all of the entities - `demo.entities.entities`. Each entity has a `props` field that contains all networked properties for that entity.

If you dump all entity props, you've pretty much dumped the entire state of the world.
